### PR TITLE
[fix](multi-catalog) verify the precision of datetime types for each data source

### PR DESCRIPTION
**Pull request by AshinGau**
# Proposed changes

Fix threes bugs of timestampv2 precision:
1. Hive catalog doesn't set the precision of timestampv2, and can't get the precision from hive metastore, so set the largest precision for timestampv2;
2. Jdbc catalog use datetimev1 to parse timestamp, and convert to timestampv2, so the precision is lost.
3. TVF doesn't use the precision from meta data of file format.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/external/jdbc/JdbcClient.java [hunk 1]
```diff
@@ -524,7 +526,8 @@ public Type mysqlTypeToDoris(JdbcFieldSchema fieldSchema) {
             case "TIMESTAMP":
             case "DATETIME":
             case "DATETIMEV2": // for jdbc catalog connecting Doris database
-                return ScalarType.createDatetimeV2Type(0);
+                // mysql can support microsecond
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/external/jdbc/JdbcClient.java:529]: Add TODO here.
We should get real precision of DATETIMEV2.

**Review by morningman — COMMENTED** [review]
Please modify document

File: be/src/vec/exec/format/parquet/schema_desc.cpp [hunk 2]
```diff
@@ -240,6 +240,15 @@ TypeDescriptor FieldDescriptor::convert_to_doris_type(tparquet::LogicalType logi
         type = TypeDescriptor(TYPE_TIMEV2);
     } else if (logicalType.__isset.TIMESTAMP) {
         type = TypeDescriptor(TYPE_DATETIMEV2);
+        const auto& time_unit = logicalType.TIMESTAMP.unit;
+        if (time_unit.__isset.MILLIS) {
+            type.scale = 3;
+        } else if (time_unit.__isset.MICROS) {
+            type.scale = 6;
+        } else if (time_unit.__isset.NANOS) {
```
[the remark this answers is not in the document]
Comment (AshinGau) [reply be/src/vec/exec/format/parquet/schema_desc.cpp:248]: done

File: fe/fe-core/src/main/java/org/apache/doris/external/jdbc/JdbcClient.java (hunk 1 above)
Comment (AshinGau) [reply fe/fe-core/src/main/java/org/apache/doris/external/jdbc/JdbcClient.java:529]: done

**Comment by AshinGau** [comment]
run buildall

**Comment by AshinGau** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**